x-pack/legacy/plugins/ingest/server/libs/policy.ts

```diff
@@ -4,268 +4,248 @@
+      page: 1,
+      perPage: 25,
+    }
+  ): Promise<{ items: Policy[]; total: number; page: number; perPage: number }> {
+    const response = await this.adapter.list(user, options);
+
+    const dataSourcesIds = unique(flatten(response.items.map(policy => policy.datasources || [])));
 
-  public async listVersions(
-    sharedID: string,
-    activeOnly = true,
-    page: number = 1,
-    perPage: number = 25
-  ): Promise<PolicyFile[]> {
-    const policys = await this.adapter.listVersions(sharedID, activeOnly, page, perPage);
-    return policys;
+    const datasources: Datasource[] = await this.libs.datasources.getByIDs(user, dataSourcesIds);
+
+    return {
+      ...response,
+      items: response.items.map(policy => {
+        return {
+          ...policy,
+          datasources: (policy.datasources || []).map(id => {
+            return datasources.find(ds => ds.id === id);
+          }),
+        } as Policy;
+      }),
+    };
   }
 
+  // public async changeLog(
+  //   id: string,
+  //   page: number = 1,
+  //   perPage: number = 25
+  // ): Promise<PolicyFile[]> {
+  //   const policys = await this.adapter.listVersions(sharedID, activeOnly, page, perPage);
+  //   return policys;
+  // }
+
   public async update(
+    user: FrameworkUser,
     id: string,
-    policy: Partial<{
-      name: string;
-      description: string;
-      monitoring_enabled: boolean;
-    }>
-  ): Promise<{ id: string; version: number; shared_id: string }> {
-    const invalidKeys = Object.keys(policy).filter(
-      key => !['name', 'description', 'monitoring_enabled'].includes(key)
+    policy: Partial<StoredPolicy>
+  ): Promise<StoredPolicy> {
+    const invalidKeys = Object.keys(policy).filter(key =>
+      ['updated_on', 'updated_by'].includes(key)
     );
 
     if (invalidKeys.length !== 0) {
       throw new Error(
         `Update was called with policy paramaters that are not allowed: ${invalidKeys}`
       );
     }
-    const oldPolicy = await this.adapter.get(id);
-
-    if (oldPolicy.status !== 'active') {
-      throw new Error(
-        `Policy ${oldPolicy.id} can not be updated becuase it is ${oldPolicy.status}`
-      );
-    }
-
-    const newPolicy = await this._update(oldPolicy, policy);
-    return newPolicy;
-  }
+    const oldPolicy = await this.adapter.get(user, id);
 
-  public async delete(sharedId: string): Promise<{ success: boolean }> {
-    if (sharedId === DEFAULT_POLICY_ID) {
-      throw new Error('Not allowed (impossible to delete default policy)');
+    if (!oldPolicy) {
+      throw new Error('Policy not found');
     }
-    // TODO Low priority - page through vs one large query as this will break if there are more then 10k past versions
-    const versions = await this.listVersions(sharedId, false, 1, 10000);
 
-    // TODO bulk delete
-    for (const version of versions) {
-      await this.adapter.deleteVersion(version.id);
+    if (oldPolicy.status === Status.Inactive && policy.status !== Status.Active) {
```

Comment:
If the policy is inactive, and this change does not make the policy active, don't allow updates. If we do it could lead to "but the last time I used this policy, it worked!"